src/utils/focus.rs

```diff
@@ -45,9 +46,16 @@ impl<BackendData: Backend> PointerTarget<MagmaState<BackendData>> for FocusTarge
         event: &MotionEvent,
     ) {
         match self {
-            FocusTarget::Window(w) => {
-                PointerTarget::enter(&w.wl_surface().unwrap(), seat, data, event)
-            }
+            FocusTarget::Window(w) => match w.underlying_surface() {
+                WindowSurface::Wayland(w) => {
+                    PointerTarget::enter(w.wl_surface(), seat, data, event)
+                }
+                #[cfg(feature = "xwayland")]
+                WindowSurface::X11(x) => match x.wl_surface() {
+                    Some(w) => PointerTarget::enter(&w, seat, data, event),
+                    None => debug!("Pointer entered non-visible xwayland window"),
```

Comment:
what are non visible xwayland windows

Reply:
X11 clients that do not have an associated wl_surface. This seems to occur when the window is being closed. The debug statements are probably unnecessary.